prog/rand.go

```diff
@@ -66,13 +66,21 @@ var specialInts = []uint64{
 	(1 << 32) - 1, (1 << 32), (1 << 32) + 1,
 }
 
-func (r *randGen) randInt() uint64 {
-	v := r.rand64()
+func (r *randGen) randInt64() uint64 {
+	return r.randInt(8)
+}
+
+func (r *randGen) randInt(size uint64) uint64 {
+	if size > 8 {
```

Comment:
This still makes me uncomfortable. The larger sizes used for string-formatted integers mean completely different thing, consider they are in different units. Different units must never be compared. It's like passing kilograms as distance and then trying to workaround the resulting chaos.
E.g. later we may have string-formatted integers that occupy less space (<=8), e.g. 4, and then this code will silently break because that 4 is still size of the resulting string, not size of the binary-represented integer. Most likely no existing tests will catch that. This is fragile.
Maybe we need a new function on int type that returns the size that you need here. That function could look at the format and return 8 for string formatted arguments with a comment as to why we are doing this and how it can be improved in future.